File: app/src/main/java/com/z/palette/adapter/RecyclerViewAdapter.java
package com.z.palette.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.z.palette.BitmapUtils;
import com.z.palette.R;

import java.util.List;

/**
 * @author zhangkb
 */
public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder> {
  //  private List<Palette.Swatch> swatchList;
    private List<PaletteColorsBean> paletteColorsBeans;

    public RecyclerViewAdapter(List<PaletteColorsBean> paletteColorsBeans) {
        this.paletteColorsBeans = paletteColorsBeans;
    }

    Context context;

    @NonNull
    @Override
    public RecyclerViewAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        this.context=viewGroup.getContext();
        return new RecyclerViewAdapter.ViewHolder(LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.adapter_main, viewGroup, false));
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerViewAdapter.ViewHolder viewHolder, int i) {

        if(null!=paletteColorsBeans.get(i).getColor()){
            viewHolder.mView.setImageBitmap(null);
            viewHolder.mView.setBackgroundColor(paletteColorsBeans.get(i).getColor().getRgb());
            viewHolder.mText.setText(paletteColorsBeans.get(i).getColorText());
        }else {
            viewHolder.mText.setText(" ");
            viewHolder.mView.setImageBitmap(paletteColorsBeans.get(i).getBitmap());
        }
        viewHolder.mView.setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View v) {

                if(null!=paletteColorsBeans.get(i).getBitmap()){
                    BitmapUtils.saveImage(paletteColorsBeans.get(i).getBitmap(),context);
                    Toast.makeText(context,"保存成功！",Toast.LENGTH_SHORT).show();
                }

                return false;
            }
        });
    }

    @Override
    public int getItemCount() {
        return paletteColorsBeans.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private ImageView mView;
        private TextView mText;

        public ViewHolder(View itemView) {
            super(itemView);
            mView = itemView.findViewById(R.id.image);
            mText = itemView.findViewById(R.id.text);
        }
    }

}

File: app/src/main/java/com/z/palette/adapter/RecyclerViewAdapter.kt
package com.z.palette.adapter

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.z.palette.BitmapUtils
import com.z.palette.R

/**
 * @author zhangkb
 */
class RecyclerViewAdapter(//  private List<Palette.Swatch> swatchList;
        private val paletteColorsBeans: List<PaletteColorsBean>) : RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder>() {
    var context: Context? = null
    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {
        context = viewGroup.context
        return ViewHolder(LayoutInflater.from(viewGroup.context).inflate(R.layout.adapter_main, viewGroup, false))
    }

    private var mHeight: Int = 0;
    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) {
        if (null != paletteColorsBeans[i].color) {
            viewHolder.mColor.visibility = View.VISIBLE
            viewHolder.mView.visibility = View.GONE
            viewHolder.mColor.setBackgroundColor(paletteColorsBeans[i].color.rgb)
            viewHolder.mText.text = paletteColorsBeans[i].colorText
        } else {

            viewHolder.mColor.visibility = View.GONE
            viewHolder.mView.visibility = View.VISIBLE
            //viewHolder.mView.maxHeight=paletteColorsBeans[i].bitmap.
            context?.let { Glide.with(it).load((paletteColorsBeans[i].bitmap)).into(viewHolder.mView) }
            //viewHolder.mView.setImageBitmap(paletteColorsBeans[i].bitmap)
        }
        viewHolder.mColor.setOnLongClickListener {
            //获取剪贴板管理器：
            val cm = context?.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
            val mClipData = ClipData.newPlainText("Label", paletteColorsBeans[i].colorText)
            cm.setPrimaryClip(mClipData)
            Toast.makeText(context, "已复制到剪切板！", Toast.LENGTH_SHORT).show()
            false
        }
        viewHolder.mView.setOnLongClickListener {
            if (null != paletteColorsBeans[i].bitmap) {

                //获取剪贴板管理器：
                val cm = context?.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
                val mClipData = ClipData.newPlainText("Label", paletteColorsBeans[i].colorText)
                cm.setPrimaryClip(mClipData);

                BitmapUtils.saveImage(paletteColorsBeans[i].bitmap, context)
                Toast.makeText(context, "保存成功！", Toast.LENGTH_SHORT).show()
            }
            false
        }
    }

    override fun getItemCount(): Int {
        return paletteColorsBeans.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        public val mView: ImageView
        public val mColor: ImageView
        public val mText: TextView

        init {
            mView = itemView.findViewById(R.id.image)
            mText = itemView.findViewById(R.id.text)
            mColor = itemView.findViewById(R.id.imageColor)
        }
    }

}